examples/predictors/lstm/xprize_predictor.py

```diff
@@ -91,82 +92,82 @@ def __init__(self, path_to_model_weights, data_url, cutoff_date_str):
-            initial_total_cases = prev_confirmed_cases[-1]
-            pop_size = np.array(cdf.Population)[-1]  # Population size doesn't change over time
-
-            # Compute predictor's forecast
-            pred_total_cases, pred_new_cases = self._convert_ratios_to_total_cases(
-                preds,
-                WINDOW_SIZE,
-                prev_new_cases,
-                initial_total_cases,
-                pop_size)
-            # OPTIONAL: Smooth out pred_new_cases
-            # # If window size is 7, take the previous 6 new cases so we start doing a 7 day moving average for
-            # # the first pred new cases
-            # temp_pred_new_cases = list(prev_new_cases[-(WINDOW_SIZE-1):]) + pred_new_cases
-            # smooth_pred_new_cases = self._smooth_case_list(temp_pred_new_cases, WINDOW_SIZE)
-            # # Get rid of the first window_size - 1 NaN values where
-            # # there was not enough data to compute a moving average
-            # pred_new_cases = smooth_pred_new_cases[WINDOW_SIZE-1:]
+        # For each requested geo
+        geos = npis_df.GeoID.unique()
+        for g in geos:
+            cdf = self.df[self.df.GeoID == g]
+            if len(cdf) == 0:
+                # we don't have historical data for this geo: return zeroes
+                pred_new_cases = [0] * nb_days
+                geo_start_date = start_date
+            else:
+                last_known_date = cdf.Date.max()
+                # Start predicting from start_date, unless there's a gap since last known date
+                geo_start_date = min(last_known_date + np.timedelta64(1, 'D'), start_date)
+                npis_gdf = npis_df[(npis_df.Date >= geo_start_date) & (npis_df.Date <= end_date)]
+
+                pred_new_cases = self._get_new_cases_preds(cdf, g, npis_gdf)
 
             # Append forecast data to results to return
+            country = npis_df[npis_df.GeoID == g].iloc[0].CountryName
+            region = npis_df[npis_df.GeoID == g].iloc[0].RegionName
             for i, pred in enumerate(pred_new_cases):
-                # Split CountryName back into CountryName and RegionName
-                c_split = c.split(" / ")
-                country = c_split[0]
-                region = c_split[1] if len(c_split) > 1 else np.NaN
                 forecast["CountryName"].append(country)
                 forecast["RegionName"].append(region)
-                current_date = start_date + pd.offsets.Day(i)
+                current_date = geo_start_date + pd.offsets.Day(i)
                 forecast["Date"].append(current_date)
-                # forecast["ConfirmedCases"].append(pred_total_cases[i])
                 forecast["PredictedDailyNewCases"].append(pred)
 
         forecast_df = pd.DataFrame.from_dict(forecast)
-        return forecast_df
+        # Return only the requested predictions
+        return forecast_df[(forecast_df.Date >= start_date) & (forecast_df.Date <= end_date)]
```

Comment:
... and we truncate here to return only the requested rows: rows between start_date and end_date, included.